_posts/2019-09-20-knative.md

```diff
@@ -0,0 +1,159 @@
+author: sjwoodman
+---
+
+[Knative](https://knative.dev/) is a platform for running Serverless workloads on Kubernetes clusters. 
+It provides middleware abstractions and APIs that ease the development and deployment lifecycle in order to allow developers to focus on application logic.
+For example, Knative deals with the networking and exposing of applications, tracking of versions and scaling (including to zero).
+Further, it offers declarative ways to bind applications to events and data streams including those based on Apache Kafka.
+
+<!--more-->
+
+Apache Kafka can be used in Knative applications in [two ways](https://github.com/knative/eventing-contrib/tree/master/kafka): firstly, Kafka can be used as the underlying transport for internal Channels; and secondly, it can be used as the source of events.
+This blog post will focus on the latter and demonstrate how to develop a scalable application that responds to messages on a Kafka Topic.   
+The Knative project includes two components, Knative Serving and Knative Eventing.
+Knative Serving deals with the deployment, routing and scaling of the serverless containers.
+Knative Eventing is responsible for the messaging infrastructure to connect these containers to sources of events.
+
+# Installation
+
+Our application will run in Minikube although Knative and Strimzi can both work in any Kubernetes cluster - there is no vendor lock in.
+Initially we need a working Minikube cluster with the relevant components installed.
+The community documentation describes how to [install Knative on Minikube](https://knative.dev/docs/install/knative-with-minikube/).
+Notice that Knative requires a lot of resources - at least 10GB of memory and 6 CPUs are recommended.
+The installation of Knative is complicated and involves a number of steps. 
+For instance, first a ServiceMesh such as Istio, Ambassador or Gloo must be installed, then Knative Serving and finally Knative Eventing.
+
+Fortunately, one of the [contributors to Knative Eventing](http://github.com/matzew/) has developed a much simpler install script for Minishift.
+This script will install not just Knative but all the dependencies and even Strimzi itself.
+Knative version 0.8.0 will be used along with the latest release of Strimzi (currently 0.13.0).
+Checkout the [repository](https://github.com/matzew/lberk-misc/tree/eventing_080).
+It may be necessary to update some of the configuration in the `setup.sh` such as the `vm-driver` if you do not have an existing Minikube configuration.
+You should be able to use `kvm2` on Linux and `hyperkit` on MacOS.
+
+First, run the `setup.sh` script which will start the installation.
+This will take some time and sometimes fails due to a race condition when installing some of the CRDs.
+```bash
+error: unable to recognize "STDIN": no matches for kind "Image" in version "caching.internal.knative.dev/v1alpha1"
+```
+
+If that failure does occur run the `hack.sh` script which will complete the installation. 
```

Comment:
so, the hack is not needed, with the latest greatest

Reply:
Updated